pipenv/project.py

```diff
@@ -663,11 +667,68 @@ def requirements_location(self) -> str | None:
             self._requirements_location = loc
         return self._requirements_location
 
+    def _acquire_file_lock(self, file_obj):
+        """Acquire lock on an existing file object"""
+        if sys.platform == "win32":
+            import msvcrt
+
+            # Try to lock for a maximum of 10 seconds
+            start_time = time.time()
+            while (time.time() - start_time) < 10:
+                try:
+                    msvcrt.locking(file_obj.fileno(), msvcrt.LK_NBLCK, 1)
+                    return True
+                except OSError:  # noqa: PERF203
+                    time.sleep(0.1)
+            return False
+        else:
+            import fcntl
+
+            try:
+                # Use non-blocking to prevent deadlocks
+                fcntl.flock(file_obj.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+                return True
+            except OSError:
+                return False
+
+    def _release_file_lock(self, file_obj):
+        """Release lock on an existing file object"""
+        if sys.platform == "win32":
+            import msvcrt
```

Comment:
This import block is repeated in _acquire_file_lock and in _relase lock. Consider moving to the top.